app/Http/Controllers/Bible/BibleFilesetsDownloadController.php

```diff
@@ -101,20 +101,27 @@ function () use ($fileset_id, $type) {
             );
         }
 
-        $cache_params = $this->removeSpaceFromCacheParameters([$fileset_id, $book_id, $chapter, $key, $limit, $type]);
+        $bulk_access_control = $this->allowedForDownload($fileset);
```

Comment:
@bradflood , I have moved this line from the below cacheRemember method, because we need to make sure that if api_key user doesn't allowed to download content, DBP doesn't add entry to user_download entity.